task/control-plane-agent/src/mgs_common.rs

```diff
@@ -143,6 +147,108 @@ impl MgsCommon {
             CabooseError::NoImageHeader => panic!(),
         })
     }
+
+    pub(crate) fn reset_component_prepare(
+        &mut self,
+        component: SpComponent,
+    ) -> Result<(), SpError> {
+        // TODO: Add some kind of auth check before performing a reset.
+        // https://github.com/oxidecomputer/hubris/issues/723
+        ringbuf_entry!(Log::MgsMessage(MgsMessage::ResetPrepare));
+        self.reset_component_requested = component;
+        Ok(())
+    }
+
+    pub(crate) fn reset_component_trigger(
+        &mut self,
+        update: &SpUpdate,
+        component: SpComponent,
+        slot: Option<u16>,
+        intent: ResetIntent,
+        auth_data: &[u8],
+    ) -> Result<(), SpError> {
+        // TODO: Add some kind of auth check before performing a reset.
+        // https://github.com/oxidecomputer/hubris/issues/723
+        if self.reset_component_requested != component {
+            return Err(SpError::ResetComponentTriggerWithoutPrepare);
+        }
+        // If we are not resetting the SP_ITSELF, then we may come back here.
+        self.reset_component_requested.id.fill(0);
+
+        // For now, resetting the SP through reset_component() is
+        // the same as through reset()
```

Comment:
"For now" -- are we planning a change in the future that will cause this to no longer be the case? What change is it, so I know when to make it as a maintenance programmer?